### PR TITLE
Access Level Model For WSO2 Dashboard Server

### PR DESCRIPTION
**Pull request by NisalaNiroshana**

### CODE REVIEW

File: apps/portal/js/store-manager.js [hunk 1]
```diff
@@ -86,14 +86,21 @@ var getAsset, getAssets, addAsset, deleteAsset, getDashboardsFromRegistry;
 
         if (dashboards) {
             dashboards.forEach(function (dashboard) {
-                allDashboards.push(JSON.parse(registry.content(dashboard)));
+                var ContentDashboardJSON = JSON.parse(registry.content(dashboard));
```
Comment (TanyaM) [inline apps/portal/js/store-manager.js]: replace JSON.parse with parse, since this is a back-end file

File: apps/portal/js/store-manager.js [hunk 2]
```diff
@@ -86,14 +86,21 @@ var getAsset, getAssets, addAsset, deleteAsset, getDashboardsFromRegistry;
...
+                }
+                allDashboards.push(ContentDashboardJSON);
             });
         }
         if (superTenantDashboards) {
             utils.startTenantFlow(carbon.server.superTenant.tenantId);
             superTenantDashboards.forEach(function (dashboard) {
                 var parsedDashboards = JSON.parse(superTenantRegistry.content(dashboard));
                 if (parsedDashboards.shareDashboard) {
+                    if (stringify(parsedDashboards.permissions.owners) === "null") {
```
Comment (TanyaM) [inline apps/portal/js/store-manager.js]: use hasOwnProperty

File: apps/portal/controllers/dashboards.jag [hunk 3]
```diff
@@ -148,8 +148,9 @@ var fresh;
             title: request.getParameter('title'),
             description: request.getParameter('description'),
             permissions: {
-                viewers: ['Internal/everyone'],
-                editors: ['Internal/everyone']
+                viewers: ["Internal/" + dashboardId + "-viewer"],
```
Comment (sinthuja) [inline apps/portal/controllers/dashboards.jag]: Specify the 'Internal',  "viewer", etc strings as constants.

File: apps/portal/js/store-manager.js (hunk 1 above)
Comment (sinthuja) [inline apps/portal/js/store-manager.js]: The variable name should be starting with lowercase and camel case.

File: components/dashboard-deployment/src/main/java/org/wso2/carbon/dashboard/authorization/util/AuthorizationUtil.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    /**
+     * @param tenantId   Tenant ID of the user
+     * @param username   User name of the user
+     * @param permission Required permission to access the particular component
+     * @return true if the user has the required permission, unless false
+     * @throws UserStoreException,RegistryException
+     */
+    public static boolean isUserAuthorized(int tenantId, String username, String permission)
+            throws UserStoreException, RegistryException {
+
```
Comment (sinthuja) [inline components/dashboard-deployment/src/main/java/org/wso2/carbon/dashboard/authorization/util/AuthorizationUtil.java]: Remove the blank line.

File: components/dashboard-deployment/src/main/resources/META-INF/component.xml [hunk 5]
```diff
@@ -17,7 +17,7 @@
 -->
 
 <component xmlns="http://products.wso2.org/carbon">
-	<!-- <deployers>
+    <!-- <deployers>
```
Comment (sinthuja) [inline components/dashboard-deployment/src/main/resources/META-INF/component.xml]: Remove the commented deployer segment.

**Pull request merged**